test/runtests.jl

```diff
@@ -71,6 +71,72 @@ using Random # for `randstring`
         rm(fname1, force=true)
         rm(fname2, force=true)
     end
+
+    @testset "non-allocating read" begin
+        fname1 = tempname() * ".fits"
+        try
```

Comment:
Note that instead of using `try`/`finally` you can open the FITS file with
```julia
FITS(fname, "w") do f
    # ...
end
```
which will automatically close it at the end.  You can find some examples in this file.

The `try`/`finally` that you still see here are leftovers of the times where this `FITS` method was not available, yet.

Reply:
I have updated the tests to use this method

Reply:
@kbarbary there is `rm(fname1, force=true)` after the `FITS ... do ... end` block.

Reply:
The `try finally` might still be necessary to clean up if there's an error, in which case  the `rm` won't get called. I have fixed this in the latest commit